fdbserver/DataDistribution.actor.cpp

```diff
@@ -322,10 +322,12 @@ ACTOR Future<Void> debugCheckCoalescing(Database cx) {
 }
 
 struct DataDistributor;
-void runAuditStorage(Reference<DataDistributor> self,
-                     AuditStorageState auditStates,
-                     int retryCount,
-                     DDAuditContext context);
+void runAuditStorage(
+    Reference<DataDistributor> self,
+    AuditStorageState auditStates,
+    int retryCount,
+    DDAuditContext context,
+    Optional<std::unordered_map<UID, bool>> serverProgressFinishMap = Optional<std::unordered_map<UID, bool>>());
```

Comment:
Where is `serverProgressFinishMap` used?

Reply:
serverProgressFinishMap is used when check whether a server completes the check. If yes, we do not want to repeatedly check a complete server. To achieve so, we use serverProgressFinishMap to record the servers that have been checked.